### PR TITLE
Small CMake improvements

### PR DESCRIPTION
**Pull request by carstene1ns**
Makes CMake build method on-par with autotools.
Fixes the wrong installation folders and pkg-config file generation.
Moves the CMakeLists.txt file to the root folder. Initial work on #215.

No functional changes.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,104 +1,102 @@
...
-
+# C++11 support is needed
 include(CheckCXXCompilerFlag)
 CHECK_CXX_COMPILER_FLAG("-std=c++11" COMPILER_SUPPORTS_CXX11)
 if(COMPILER_SUPPORTS_CXX11)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 else()
-    message(STATUS "The compiler ${CMAKE_CXX_COMPILER} has no C++11 support.")
+  message(ERROR "The compiler ${CMAKE_CXX_COMPILER} has no C++11 support.")
```
Comment (carstene1ns) [inline CMakeLists.txt]: upps. `FATAL_ERROR` - https://cmake.org/cmake/help/v3.0/command/message.html

**Comment by Ghabry** [comment]
The move of the file will need an update to the CMakeLists.txt of Player because of the PLAYER_BUILD_LIBLCF functionality.

**Comment by carstene1ns** [comment]
k, going to change https://github.com/EasyRPG/Player/blob/master/CMakeLists.txt#L238 then.

**Comment by Ghabry** [comment]
#NeedsRebase ;)

**Comment by carstene1ns** [comment]
hnng

**Comment by carstene1ns** [comment]
hmm, somehow git does not recognize the rename after the rebase ....

**Comment by carstene1ns** [comment]
Made it 2 commits now, better diff view: https://github.com/EasyRPG/liblcf/pull/217/commits/6338d833949adfb31283c783bd8bc23f2f019034.

**Pull request merged**